target es2021 (#23307)
microsoft/FluidFramework: .changeset/ripe-taxes-press.md

---
---
---
"section": other
---

Target ES2021

> The TypeScript build has been updated to target ES2021 instead of ES2020.

Reviewer: ```suggestion
The TypeScript build for Fluid Framework packages has been updated to target ES2021 instead of ES2020.
```

This may result in newer JavaScript language features being used.

> This does not include updating the library version being used.

Reviewer: The differences here may not be obvious. Any ideas to clarify? Maybe calling out that this just changes the JS that TypeScript outputs for our packages, while lib is the available APIs you can use when writing TS source? Ugh that might be more confusing.

Author: ```suggestion
This does not change TypeScript types, nor the JavaScript APIs being used.
```

We only support users which support ES2022, so updating to target ES2021 should not break any supported use-case.

> Any users which do not have at least ES2021 language feature support may need transpile out some additional cases after this change.

Reviewer: ```suggestion
Any users which do not have at least ES2021 language feature support may need to transpile out some additional cases after this change.
```

> This should result is slightly reduced bundle size and slightly improved performance for users not transpiling these features out.

Reviewer 2: ```suggestion
This should result in slightly reduced bundle size and slightly improved performance for users not transpiling these features out.
```

> No major changes are expected.

Reviewer: ```suggestion
No major impact is expected.
```
